src/printer.js

```diff
@@ -2324,6 +2325,27 @@ function genericPrintNoParens(path, options, print, args) {
         join(concat([line, "| "]), printed)
       ]);
 
+      let hasParens;
+
+      if (n.type === "TSUnionType") {
```

Comment:
Not really happy with this, but it seemed better to keep this logic here than spread it into `TSParenthesizedType`?

Reply:
Feels about the same either way to me personally